src/java/com/github/ruleant/getback_gps/AbstractGetBackGpsActivity.java

```diff
@@ -411,7 +472,7 @@ public final void renameDestination() {
         // Get the EditText object containing the location name
         final EditText etLocationName
                 = (EditText) dialogView.findViewById(R.id.location_name);
-        if (etLocationName != null) {
+        if (etLocationName != null && mService != null && mService.getDestination() != null) {
```

Comment:
nice find! Without the check if mService exists, the app crashes. :p

It appears that the service doesn't exist yet, when this dialog is reinstated after an orientation change, so the current destination name isn't filled in.
If the name was changed, the changed name is rememered in the state and does appear in the dialog, so I guess the statetracking doesn't store the initial value.
(Small corner case which doesn't happen with all the other dialog entries, because the initial state is usually empty).

Reply:
After looking at the code for StateTrackingDialogBuilder it looks to me that this handles the initial state and the changes well, But that the 'afterTextChanged()' is only triggered when the value of etLocationName is changed in the UI, and not if the text is changed in the code.

Perhaps we'll have to change the state as well, when this text is updated in the code? What do you think?

Reply:
Also thank you very much for your feedback. This way it's fun to contribute something to the project. :-)